libs/cart/src/effects/cart-item.effects.spec.ts

```diff
@@ -168,15 +194,47 @@ describe('Daffodil | Cart | CartItemEffects', () => {
     });
 
     describe('and the call to CartItemService is successful', () => {
-      beforeEach(() => {
-        mockCart.items.push(mockCartItem);
-        daffCartItemDriverSpy.add.and.returnValue(of(mockCart));
-        const cartItemAddSuccessAction = new DaffCartItemAddSuccess(mockCart);
-        actions$ = hot('--a', { a: cartItemAddAction });
-        expected = cold('--b', { b: cartItemAddSuccessAction });
+
+			it(`should dispatch a CartItemAddSuccess action with cart items of the correct state 
+					when a new product is added to the cart`, () => {
+				store.overrideSelector(selectAllCartItems, []);
+				const newCart = {
+					...mockCart,
+					items: [
+						mockCartItem
+					]
+				}
+				daffCartItemDriverSpy.add.and.returnValue(of(newCart));
+				const cartItemAddSuccessAction = new DaffCartItemAddSuccess({
+					...newCart,
+					items: [
+						{ ...mockCartItem, state: DaffCartItemStateEnum.New }
+					]
+				});
+				actions$ = hot('--a', { a: cartItemAddAction });
+				expected = cold('--b', { b: cartItemAddSuccessAction });
+        expect(effects.add$).toBeObservable(expected);
       });
 
-      it('should dispatch a CartItemAddSuccess action', () => {
+			it(`should dispatch a CartItemAddSuccess action with cart items of the correct state 
+					when an existing product on the cart is added to the cart`, () => {
+				store.overrideSelector(selectAllCartItems, [mockCartItem]);
+				const newCart = {
+					...mockCart,
+					items: [{
+						...mockCartItem,
+						qty: mockCartItem.qty + 1
+					}]
+				};
+				daffCartItemDriverSpy.add.and.returnValue(of(newCart));
+				const cartItemAddSuccessAction = new DaffCartItemAddSuccess({
+					...newCart,
+					items: [
+						{ ...newCart.items[0], state: DaffCartItemStateEnum.New }
```

Comment:
I think it should show as updated when you add the same item to your cart. Isn't that effectively the same thing as updating the quantity?

Reply:
That's a good question. I think it comes down to what takes precedence: what actually changes between the old state and new state, or which process is used to "update to qty". At first I agreed with you, but then I thought that maybe how the customer begins the process (in this case by adding a product to the cart instead of changing the quantity of a cart item) should determine the state of the cart item. Maybe it should have a "New" state even though all that essentially happened was an "update", because the customer will expect to have added a product to the cart instead of updating a cart item? What do you think?

Reply:
I think the state change should be what determines updated vs. new as opposed to the process that generated it. As a customer, I would only expect to see a "new" notification when I added an item that was previously not in my cart.

That brings up some edge cases though. Consider the case where a customer has already added a composite product to their cart with required options A and B and nothing else. The customer then adds the same composite with required options A and B and optional option C. Does the cart item show 2 discrete items, one with A,B and the other with A,B,C or a single item with 2A,2B,C? And in this case, should that be new or updated?

Reply:
>  As a customer, I would only expect to see a "new" notification when I added an item that was previously not in my cart.

@damienwebdev I would like your opinion on this UX.

Reply:
In the composite product example, it would always show as a new cart item, since the "new" status is based on cart item ids and a composite product with different options is always a new cart item on the backend

Reply:
I was thinking about this and I think you're right. I think it should depend on state change rather than customer intention. On the one hand, I don't think that you would add a whole other product just to update the qty of the cart item. You'd probably just update the cart item qty in the cart. But on the other hand, I was thinking that the only time you would add a product to the cart like that is if you'd forgotten you had already added the item. In that case, it would be more useful to show an "updated" state rather than a "new" state, because the "new" state might cause you to accidentally buy two of the same product unintentionally, while showing an "updated" state might signal to you that something unexpected is going on; i.e. you already had the item in your cart.

Reply:
I changed it to what you proposed. We can wait to hear from Damien if you'd like though

Reply:
Yeah I'd say give Damien a chance to look at it but the code lgtm

Reply:
I think the state isn't quite correct as "new". My original intent here is more of a "HEY YOU, THIS THING CHANGED" kind of state indicator.

E.g. if anything changed, we need to let the user know about it.

Reply:
I think we talked about this and resolved it. Let me know if that's not the case